spras/evaluation.py

```diff
@@ -82,120 +80,159 @@ def load_files_from_dict(self, gold_standard_dict: Dict):
+                                             output_file: str):
         """
-        Takes in a node ensemble for specific dataset or specific algorithm in a dataset, and an associated gold standard node table.
-        Plots a precision and recall curve for the node ensemble against its associated gold standard node table
-        Returns output back to output_png
-        @param node_ensemble: the pre-computed node_ensemble
-        @param node_table: the gold standard nodes
-        @param output_png: the filename to save the precision and recall curves
-        @param output_file: the filename to save the precision, recall, and threshold values
+        Plots precision-recall (PR) curves for a set of node ensembles evaluated against a gold standard.
+
+        Takes in a dictionary containing either algorithm-specific node ensembles or an aggregated node ensemble
+        for a given dataset, along with the corresponding gold standard node table. Computes PC curves for
+        each ensemble and plots all curves on a single figure.
+
+        @param node_ensembles: dict of the pre-computed node_ensemble(s)
+        @param node_table: gold standard nodes
+        @param output_png: filename to save the precision and recall curves
+        @param output_file: filename to save the precision, recall, threshold values, average precision, AUC, and baseline precision
         """
         gold_standard_nodes = set(node_table['NODEID'])
 
+        # make color palette per ensemble label name
         label_names = list(node_ensembles)
         color_palette = create_palette(label_names)
 
         plt.figure(figsize=(10, 7))
 
-        dfs = []
+        prc_dfs = []
+        metric_dfs = []
 
         for label, node_ensemble in node_ensembles.items():
-
             if not node_ensemble.empty:
                 y_true = [1 if node in gold_standard_nodes else 0 for node in node_ensemble['Node']]
                 y_scores = node_ensemble['Frequency'].tolist()
                 precision, recall, thresholds = precision_recall_curve(y_true, y_scores)
                 # avg precision summarizes a precision-recall curve as the weighted mean of precisions achieved at each threshold
                 avg_precision = average_precision_score(y_true, y_scores)
-                # the (number positives)/(number instances)
-                baseline_precision = np.sum(y_true) / len(y_true)
+                auc_score = auc(recall, precision)
 
-                plt.plot(recall, precision, color=color_palette[label], marker='o', label=f'{label} PR curve (AP:{avg_precision:.4f})')
-                plt.axhline(y=baseline_precision, color=color_palette[label], linestyle='--', label=f'{label} Baseline Precision: {baseline_precision:.4f}')
+                plt.plot(recall, precision, color=color_palette[label], marker='o',
+                         label=f'{label.capitalize()} PR curve (AP: {avg_precision:.4f})')
```

Comment:
The " PR curve" part clutters the legend. We can drop it and leave only the method name and the AP.